File: Factorial.cs
using System;

namespace Factorial
{
    class Program
    {
        static void Main()
        {
            Console.Write("Enter a positive integer: ");
            int number = int.Parse(Console.ReadLine());

            if (number >= 0)
            {
                int factorial = 1;
                int i = 1;

                while (i <= number)
                {
                    factorial *= i;
                    i++;
                }

                Console.WriteLine("Factorial of " + number + " is " + factorial);
            }
            else
            {
                Console.WriteLine("The number " + number + " is not a positive integer");
            }
        }
    }
}

File: For-Loop-Problems/factorial.js
let args = process.argv.slice(2);

if (args.length < 1) {
    console.log("Usage: node script.js <number>");
    process.exit(1);
}

let num = parseInt(args[0]);
if (isNaN(num) || num < 0) {
    console.log("Invalid input! Please enter a non-negative integer.");
    process.exit(1);
}
function factorial(n) {
    if (n === 0 || n === 1) return 1; 
    let fact = 1;
    for (let i = 2; i <= n; i++) {
        fact *= i;
    }
    return fact;
}
console.log(`${num}! = ${factorial(num)}`);
